torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -5,7 +5,7 @@
 
 #include <c10/core/DeviceGuard.h>
 
-#if defined(OPEN_MPI) && OPEN_MPI
+#if ((defined(OPEN_MPI) && OPEN_MPI) || (defined(MVAPICH2_NUMVERSION) && (MVAPICH2_NUMVERSION >= 20205300))) && !USE_CUDA_MPI
```

Comment:
Good idea! I added a clarifying comment.

Reply:
`(defined(OPEN_MPI) && OPEN_MPI)` is bad pattern as it will still result in the warning.
Only workaround for this would be 
```
#ifndef OPEN_MPI
#define OPEN_MPI 0
#endif
#if OPEN_MPI
```